### PR TITLE
Support DOS-like wildcards in -requires

### LINKED ISSUE #270: -version does not allow for a single version value
Running `vswhere.exe -version 17` doesn't set the version range to be `[17,18)` but rather uses the max version for the upper bound. It may be less confusing to support a single version value for `-version` and set the maximum of the range to be the same as the minimum.

### LINKED ISSUE #276: `-requires` wildcard
I would like to do something like `-requires Microsoft.VisualStudio.Component.Windows10SDK.*`, or something similar so I don't have to maintain a list of all possible SDK versions when any will suffice.  Is that possible?

### PR DESCRIPTION
**Pull request by heaths**
Fixes #276. Also resolves #270 by including link to more details on versioning syntax.

### CODE REVIEW

**Comment by heaths** [comment]
I need to merge #279 first and will then rebase this one.

File: src/vswhere.lib/CommandArgs.cpp [hunk 1]
```diff
@@ -231,6 +234,37 @@ void CommandArgs::Usage(_In_ Console& console) const
...
+
+        case L'*':
+            accumulator += L".*";
+            break;
+
+        case L'?':
+            accumulator += L".";
+            break;
+
+        default:
```
Comment (tydunkel) [inline src/vswhere.lib/CommandArgs.cpp:259]: Anything else we should escape? I think this is all we should have in package IDs that is a regex special char so we are probably good

**Review by tydunkel — APPROVED** [review]

File: src/vswhere.lib/CommandArgs.cpp (hunk 1 above)
Comment (heaths) [reply src/vswhere.lib/CommandArgs.cpp:259]: A literal `.` is all I could think about to escape, and `*` and `?` are all the wildcards we'd support that DOS does, so I think we're fine here. I couldn't think of anything else in package IDs that would be allowed e.g., `(`, `{`, `+`, etc., that would need to be escaped. IIRC, we only allowed alphanumeric + `.`, right?

Comment (heaths) [reply src/vswhere.lib/CommandArgs.cpp:259]: Looking at all IDs in a VS2022 catalog, I see nothing else we need to escape. For a breakdown:

```powershell
$catalog = gc <path> | convertfrom-json
$catalog.packages.id | foreach { $_.GetEnumerator() } | group -noelement | sort count -desc
```
```text
Count Name
----- ----
72586 .
47409 o
44286 s
35784 e
30246 t
27549 r
26777 i
18999 c
18386 a
17515 u
16168 M
12666 1
11854 l
11776 4
11529 f
11359 R
11282 C
11018 V
 9409 T
 8418 6
 8102 d
 7036 g
 6684 S
 6478 n
 5313 b
 5227 3
 5111 m
 4696 A
 4337 7
 3850 p
 3719 X
 3559 H
 2887 P
 2782 8
 2560 x
 2321 D
 2251 0
 2220 2
 1429 5
 1388 k
 1139 I
 1076 B
 1067 y
  992 W
  952 E
  876 F
  811 v
  795 N
  793 L
  756 h
  682 9
  662 O
  469 w
  376 G
  316 U
  198 j
  191 K
  183 z
  162 -
   60 q
   38 Q
   34 _
   28 J
```

**Pull request merged**